### PR TITLE
imhex: update to 1.35.3.

### PR DESCRIPTION
**Pull request by Johnnynator**
#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

File: srcpkgs/imhex/template [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
 configure_args="-DIMHEX_OFFLINE_BUILD=ON -DIMHEX_STRIP_RELEASE=OFF
  -DUSE_SYSTEM_CURL=ON -DUSE_SYSTEM_FMT=ON -DUSE_SYSTEM_LLVM=ON -DUSE_SYSTEM_CAPSTONE=ON
  -DUSE_SYSTEM_YARA=ON -DUSE_SYSTEM_NLOHMANN_JSON=ON -DIMHEX_DISABLE_UPDATE_CHECK=ON"
-hostmakedepends="pkg-config clang-tools-extra"
-makedepends="libcurl-devel fmt-devel llvm15 jansson-devel yara-devel json-c++
+hostmakedepends="pkg-config clang-tools-extra18"
+makedepends="libcurl-devel fmt-devel llvm18-devel jansson-devel yara-devel json-c++
  freetype-devel glfw-devel gtk+3-devel python3-devel file-devel mbedtls-devel
- clang-tools-extra capstone-devel"
+ clang-tools-extra18 capstone-devel"
```
Comment (Calandracas606) [inline srcpkgs/imhex/template:14]:
```suggestion
 capstone-devel"
```
`llvm18-devel` will pull in `clang-tools-extra18`

**Comment by Calandracas606** [comment]
This PR has patches to fix cross compilation: https://github.com/void-linux/void-packages/pull/50183

**Comment by Vinfall** [comment]
A bit late but [1.35.4](https://github.com/WerWolv/ImHex/releases/tag/v1.35.4) is just released before this is merged.

**Comment by Johnnynator** [comment]
> A bit late but [1.35.4](https://github.com/WerWolv/ImHex/releases/tag/v1.35.4) is just released before this is merged.

I'm aware, but 1.35.4 requires updated glfw. (Indirectly, it breaks with our current split `glfw` - `glfw-wayland` setup).
I pushed 1.35.3 first due to that.

**Comment by Vinfall** [comment]
Ah, that makes sense. Did not notice #51173 when posting, my bad.

**Pull request merged**